http3/client.go

```diff
@@ -98,32 +110,92 @@ func (c *client) dial() error {
 	}
 
 	// run the sesssion setup using 0-RTT data
-	go func() {
-		if err := c.setupSession(); err != nil {
-			c.logger.Debugf("Setting up session failed: %s", err)
-			c.session.CloseWithError(quic.ErrorCode(errorInternalError), "")
-		}
-	}()
+	// go func() {
```

Comment:
Why?

Reply:
It was causing issues in testing because of the control streams racing to connect before the request completed

Reply:
This definitely needs to be done concurrently, otherwise you'll have to wait for at least one round trip before you can send a request.